cpp/src/arrow/memory_pool.cc

```diff
@@ -404,14 +499,25 @@ Status mimalloc_memory_pool(MemoryPool** out) {
 #endif
 }
 
+static std::unique_ptr<MemoryPool> default_pool = MemoryPool::CreateDefault();
```

Comment:
This seems to be unused. Did you intend to write
```c++
MemoryPool* default_memory_pool() { return default_pool.get(); }
```
?

Reply:
Hmm, nice catch. No, it should point to one of the existing `system_pool` etc. (so that `default_pool` doesn't differ from e.g.  `jemalloc_pool` when jemalloc is enabled)